### PR TITLE
Add parenthesis to IEx.dont_display_result/0

### PR DESCRIPTION
**Pull request by eksperimental**

### CODE REVIEW

File: lib/iex/lib/iex/evaluator.ex [hunk 1]
```diff
@@ -207,7 +207,7 @@ defmodule IEx.Evaluator do
   defp handle_eval({:ok, forms}, code, line, iex_state, history, state) do
     {result, binding, env, scope} =
       :elixir.eval_forms(forms, state.binding, state.env, state.scope)
-    unless result == IEx.dont_display_result, do: io_inspect(result)
```
Comment (eksperimental) [inline lib/iex/lib/iex/evaluator.ex]: I wonder why no warning is emitted here

File: lib/iex/lib/iex/evaluator.ex [hunk 2]
```diff
@@ -207,7 +207,7 @@ defmodule IEx.Evaluator do
...
+    unless result == IEx.dont_display_result(), do: io_inspect(result)
```
Comment (eksperimental) [inline lib/iex/lib/iex/evaluator.ex]: I wonder why no warning is emitted here

**Comment by josevalim** [comment]
We do not emit warnings for zero arity calls, we emit warnings when a variable is converted into a function call. Therefore we won't ever emit warnings in the cases above. Thank you!

**Comment by eksperimental** [comment]
Oh yeah. You are right.

**Pull request closed without merging**